charts/redpanda/secrets.go

```diff
@@ -0,0 +1,759 @@
+	listenerName := "http"
+	listenerAdvertisedName := "pandaproxy"
+	redpandaConfigPart := "pandaproxy"
+	listenerVals := dot.Values.AsMap()["listeners"].(map[string]any)[listenerName].(map[string]any)
+	snippet = append(snippet,
+		``,
+		fmt.Sprintf(`LISTENER=%s`, helmette.Quote(helmette.ToJSON(map[string]any{
+			"name":    "internal",
+			"address": internalAdvertiseAddress,
+			"port":    values.Listeners.HTTP.Port,
+		}))),
+		fmt.Sprintf(`rpk redpanda config --config "$CONFIG" set %s.advertised_%s_api[0] "$LISTENER"`,
+			redpandaConfigPart,
+			listenerAdvertisedName,
+		),
+	)
+	if len(values.Listeners.HTTP.External) > 0 {
+		externalCounter := 0
+		for externalName, externalVals := range values.Listeners.HTTP.External {
+			externalCounter = externalCounter + 1
+			snippet = append(snippet,
+				``,
+				fmt.Sprintf(`ADVERTISED_%s_ADDRESSES=()`, helmette.Upper(listenerName)),
+			)
+			for _, replicaIndex := range helmette.Until(values.Statefulset.Replicas) {
+				// advertised-port for kafka
+				port := externalVals.Port // This is always defined for kafka
+				if len(externalVals.AdvertisedPorts) > 0 {
+					if len(externalVals.AdvertisedPorts) == 1 {
+						port = externalVals.AdvertisedPorts[0]
+					} else {
+						port = externalVals.AdvertisedPorts[replicaIndex]
+					}
+				}
+
+				tmplVals := map[string]any{
+					"listenerVals":             listenerVals,
+					"externalVals":             externalVals,
+					"externalName":             externalName,
+					"externalAdvertiseAddress": externalAdvertiseAddress,
+					"values":                   dot.Values.AsMap(),
+					"replicaIndex":             replicaIndex,
+					"port":                     port,
+				}
+
+				host := advertisedHostJSON(dot, externalName, port, replicaIndex)
+				// Use the advertised-host as a template
+				address := helmette.Tpl(helmette.ToJSON(host), tmplVals)
+				prefixTemplate := ptr.Deref(externalVals.PrefixTemplate, "")
+				if prefixTemplate == "" {
+					// Required because the values might not specify this, it'll ensur we see "" if it's missing.
+					prefixTemplate = helmette.Default("", values.External.PrefixTemplate)
+				}
+				snippet = append(snippet,
+					``,
+					fmt.Sprintf(`PREFIX_TEMPLATE=%s`, helmette.Quote(prefixTemplate)),
+					fmt.Sprintf(`ADVERTISED_%s_ADDRESSES+=(%s)`,
+						helmette.Upper(listenerName),
+						helmette.Quote(address),
+					),
+				)
+			}
+
+			snippet = append(snippet,
+				``,
+				fmt.Sprintf(`rpk redpanda config --config "$CONFIG" set %s.advertised_%s_api[%d] "${ADVERTISED_%s_ADDRESSES[$POD_ORDINAL]}"`,
+					redpandaConfigPart,
+					listenerAdvertisedName,
+					externalCounter,
+					helmette.Upper(listenerName),
+				),
+			)
+		}
+	}
+
+	return snippet
+}
+
+// The following from _helpers.tpm
+
+func adminInternalTLSEnabled(dot *helmette.Dot) bool {
+	values := helmette.Unwrap[Values](dot.Values)
+
+	listener := values.Listeners.Admin
+	enabled := ptr.Deref(listener.TLS.Enabled, values.TLS.Enabled)
```

Comment:
NIT: There should be option to 
```suggestion
	enabled := values.Listeners.Admin.TLS.IsEnabled(&values.TLS)
```

Reply:
Sorry wrong place